src/components/panningViews/panToDismissView.js

```diff
@@ -0,0 +1,301 @@
+import _ from 'lodash';
+import PropTypes from 'prop-types';
+import React from 'react';
+import {Animated} from 'react-native';
+import {BaseComponent} from '../../commons';
+import {Constants} from '../../helpers';
+import asPanViewConsumer from './asPanViewConsumer';
+import PanningProvider from './panningProvider';
+
+const DEFAULT_SPEED = 20;
+const DEFAULT_BOUNCINESS = 6;
+const DEFAULT_DISMISS_ANIMATION_DURATION = 280;
+
+/**
+ * @description: PanToDismissView component created to making listening to swipe and drag events easier,
+ * Has to be used as a child of a PanningProvider that also has a PanListenerView
+ */
+class PanToDismissView extends BaseComponent {
+  static displayName = 'PanToDismissView';
+  static propTypes = {
+    /**
+     * The directions of the allowed pan (default allows all directions)
+     * Types: UP, DOWN, LEFT and RIGHT (using PanningProvider.Directions.###)
+     */
+    directions: PropTypes.arrayOf(PropTypes.oneOf(Object.values(PanningProvider.Directions))),
+    /**
+     * onDismiss callback
+     */
+    onDismiss: PropTypes.func,
+    /**
+     * The animation speed (default is 20)
+     */
+    animationSpeed: PropTypes.number,
+    /**
+     * The animation bounciness (default is 6)
+     */
+    animationBounciness: PropTypes.number,
+    /**
+     * The dismiss animation duration (default is 280)
+     */
+    dismissAnimationDuration: PropTypes.number,
+  };
+
+  static defaultProps = {
+    directions: [
```

Comment:
does the order of the directions has a meaning?

Reply:
No, I use `.includes`, is that OK?